benchmarks/oss_fuzz_benchmark_integration.py

```diff
@@ -0,0 +1,243 @@
+# 2. The OSS-Fuzz code depends on absolute imports.
+# 3. The OSS-Fuzz code assumes it is run from the OSS-Fuzz directory and can
+# accdidentaly break our repo.
+OSS_FUZZ_DIR = os.path.join(utils.ROOT_DIR, 'third_party', 'oss-fuzz')
+OSS_FUZZ_REPO_PATH = os.path.join(OSS_FUZZ_DIR, 'infra')
+
+
+class BaseRepoManager:
+	"""Base repo manager."""
+
+	def __init__(self, repo_dir):
+	    self.repo_dir = repo_dir
+
+	def git(self, cmd, check_result=False):
+	    """Run a git command.
+
+	    Args:
+	      command: The git command as a list to be run.
+	      check_result: Should an exception be thrown on failed command.
+
+	    Returns:
+	      stderr, stdout, error code.
+	    """
+	    return new_process.execute(['git'] + cmd,
+	                         cwd=self.repo_dir)
+
+
+class BaseBuilderRepo:
+	"""Repo of base-builder images."""
+
+	def __init__(self):
+		self.timestamps = []
+		self.digests = []
+
+	def add_digest(self, timestamp, digest):
+		"""Add a digest."""
+		self.timestamps.append(timestamp)
+		self.digests.append(digest)
+
+	def find_digest(self, timestamp):
+		"""Find the latest image before the given timestamp."""
+		index = bisect.bisect_right(self.timestamps, timestamp)
+		if index > 0:
+			return self.digests[index - 1]
+
+		raise ValueError('Failed to find suitable base-builder.')
+
+
+def copy_oss_fuzz_files(project, commit_date, benchmark_dir):
+    """Checkout the right files from OSS-Fuzz to build the benchmark based on
+    |project| and |commit_date|. Then copy them to |benchmark_dir|."""
+    cwd = os.getcwd()
+    oss_fuzz_repo_manager = BaseRepoManager(OSS_FUZZ_DIR)
+    projects_dir = os.path.join(OSS_FUZZ_DIR, 'projects', project)
+    os.chdir(OSS_FUZZ_DIR)
+    try:
+        # Find an OSS-Fuzz commit that can be used to build the benchmark.
+        _, oss_fuzz_commit, _ = oss_fuzz_repo_manager.git([
+            'log', '--before=' + commit_date.isoformat(), '-n1', '--format=%H',
+            projects_dir
+        ], check_result=True)
+        oss_fuzz_commit = oss_fuzz_commit.strip()
+        if not oss_fuzz_commit:
+            logs.warning('No suitable earlier OSS-Fuzz commit found.')
+            return False
+        oss_fuzz_repo_manager.git(['checkout', oss_fuzz_commit, projects_dir],
+                                  check_result=True)
+        shutil.copytree(projects_dir, benchmark_dir, ignore=shutil.ignore_patterns("project.yaml"))
+        return True
+    finally:
+        oss_fuzz_repo_manager.git(['reset', '--hard'])
+        # This must be done in this order or else we reset our fuzzbench repo.
+        os.chdir(cwd)
+
+
+def get_benchmark_name(project, fuzz_target, benchmark_name=None):
+    """Returns the name of the benchmark. Returns |benchmark_name| if is set.
+    Otherwise returns a name based on |project| and |fuzz_target|."""
+    if benchmark_name is not None:
+        return benchmark_name
+    return project + '_' + fuzz_target
+
+
+def _load_base_builder_repo():
```

Comment:
can you add `docker` to this function name? `repo` is unclear on its own.

Reply:
Done